### ISSUE TITLE
[Faceted Filters] Missing Esc key support

### ISSUE THREAD

**Issue by svinkle** [Shopify/dawn#108]
## Issue summary

When navigating through the Facet options, the <kbd>Esc</kbd> key is not supported to dismiss the facet content. This may lead to a frustrating user experience as keyboard-only and screen reader users may expect this functionality. Otherwise, the user will need to navigate backward through the DOM to reach the activator control.

<details>
  <summary>Screenshots</summary>
<img width="948" alt="Screen shot of mobile faceted filters section. Facet controls are highlighted." src="https://user-images.githubusercontent.com/1392632/124605884-5e825000-de3a-11eb-98be-52f01c91b271.png">

</details>

<!-- ### Current code

#### HTML

```html

``` -->

## Steps to reproduce

1. Go to a collection page with Dawn theme with any supported browser.
2. Navigate to and activate a Filter control.
3. Press <kbd>Esc</kbd> key to dismiss.

## Behavior

### Expected

- Facet content to be hidden from view, focus to return to the activator control.

### Actual

- <kbd>Esc</kbd> key is not supported.

## Recommendation

1. Add <kbd>Esc</kbd> key support to dismiss the facet content.
2. On dismiss, shift keyboard focus back to the activator control.

With this in place, keyboard-only users and screen reader users will have focus set back to the activator control providing a quick mechanism to continue navigating from this point.

<!-- ### Recommended code

#### HTML

```html

``` -->

<!-- #### Diff

```diff

``` -->

## Specifications

- **Component:** Faceted Filters
- **WCAG Principal:** Operable
- **WCAG SC:** [2.1.1 Keyboard](https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html)
- **Severity:** Medium
- **Effort:** Medium
<!-- - **Related:** -->

**Comment by tyleralsbury** [comment]
**Possible solution**

A small JS script that checks for esc and queries for .js-filter[open] and closes it. Alternatively, if it's part of the existing filtering JS, we could keep track of the open filter on the custom element and not need to query at the moment that the user hits esc since we already have it cached.
